### DISCUSSION TITLE
Questions about two UI elements and their functions

### DISCUSSION THREAD

**Discussion by jonathanhirschman** [StreamController/StreamController#177] [Q&A] [answered]
Let me first say that this application is terrific, and thanks so much for doing it. Two questions:

For this icon: 
![image](https://github.com/user-attachments/assets/96f2285d-b8ca-4cf5-95e8-82f085b0be8e)
Sometimes it works as a key icon off/on toggle, sometimes it does nothing (as with keys that have custom images). Is this just a toggle for built-in/plugin function icons? It seems a bit unintuitive.

For this icon:
![image](https://github.com/user-attachments/assets/9741296d-13dd-47ea-9b19-3325698a5144)
So help me, I have no idea what it does, can't seem to trigger it with labels, without labels some options checked, all, none - no clue.

Thanks!

**Comment by Core447** [comment, marked as answer]
> Let me first say that this application is terrific, and thanks so much for doing it.

Thanks!

Generally speaking there are page labels/icons etc and action labels/icons etc. The page ones are always prioritized to allow the user to override action labels/icons etc.

The two buttons you're referring to have been added recently and allow users to choose which action should set the visible image and labels. This means you can have e.g. one "Analog clock" action and one "Page Change" action on the same button and via the first button, you're able to select which of the both action should show it's icon.
The same goes for the second button which does basically the same, but for the labels. The only difference besides that is that, because there are three labels (top, center, bottom) there are 3 options packed into the one button.

Something to keep in mind: If a manual (=page) image or label is set, they will always override the action ones (like mentioned above), this means the control buttons won't influence the appearance of the key.

I hope this helps a bit, if not let me know.

**Reply by jonathanhirschman** [reply]
That helps a lot; I looked for information about this on the wiki, but didn't see it. Might I suggest pop-up hints on mouse hover? Everything else in the app is pretty straightforward IMO.

Thanks so much for taking the time to answer, much appreciated.

**Reply by Core447** [reply]
You're welcome! Yeah, adding this info to the app itself makes absolutely sense! The wiki is more more plugin developers, that's why such info isn't on there

**Reply by jonathanhirschman** [reply]
If I may be so bold, it might make sense to have those buttons also affect manually set labels or images the same way, while still letting manually set objects override the same way as now. It'd be less complex, more obvious to the user, and having such functionality would have at least one (limited) use case - a place to have assets still available in case you reconfigure from combo functionality. Just a thought.

One more thought: I hope that you're going to accept donations/contributions, or alternately have a way to get paid for this. I'm suggesting this selfishly, want to see the project stick around. I'm guessing that other non-coder users feel the same way.

**Reply by Core447** [reply]
> If I may be so bold, it might make sense to have those buttons also affect manually set labels or images the same way, while still letting manually set objects override the same way as now. It'd be less complex, more obvious to the user, and having such functionality would have at least one (limited) use case - a place to have assets still available in case you reconfigure from combo functionality. Just a thought

Can you maybe share a simple sketch of this?

> One more thought: I hope that you're going to accept donations/contributions, or alternately have a way to get paid for this. I'm suggesting this selfishly, want to see the project stick around. I'm guessing that other non-coder users feel the same way.

I can add this at some point
